docs(service-worker): describe how asset-/data-group order affects request handling (#38364)
angular/angular: aio/content/guide/service-worker-config.md

```json
{
  "assetGroups": [
    {
      ...
    },
    {
      ...
    }
  ]
}
```

<div class="alert is-helpful">

> When the ServiceWorker is handling a request, it checks asset groups in the order in which they appear in `ngsw-config.json`.

Reviewer: ```suggestion
When the ServiceWorker handles a request, it checks asset groups in the order in which they appear in `ngsw-config.json`.
```

The first asset group that matches the requested resource handles the request.

> Therefore, you should put the more specific asset groups higher in the list.

Reviewer: ```suggestion
It is recommended that you put more specific asset groups higher in the list.
```
"Therefore" is a bit archaic. This rewording makes the fact that we're recommending a course of action clearer.

Unlike asset resources, data requests are not versioned along with the app. They're cached according to manually-configured policies that are more useful for situations such as API requests and other data dependencies.

This field contains an array of data groups, each of which defines a set of data resources and the policy by which they are cached.

```json
{
  "dataGroups": [
    {
      ...
    },
    {
      ...
    }
  ]
}
```

<div class="alert is-helpful">

> When the ServiceWorker is handling a request, it checks data groups in the order in which they appear in `ngsw-config.json`.

Reviewer: ```suggestion
When the ServiceWorker handles a request, it checks data groups in the order in which they appear in `ngsw-config.json`.
```

The first data group that matches the requested resource handles the request.

> Therefore, you should put the more specific data groups higher in the list.

Reviewer: ```suggestion
It is recommended that you put more specific data groups higher in the list.
```
"Therefore" is a bit archaic. This rewording makes the fact that we're recommending a course of action clearer.
